### ISSUE TITLE
[Question] Why parsing yaml to the resource duplicates kind property?

### ISSUE THREAD

**Issue by sknot-rh** [fabric8io/kubernetes-client#3195]
Hi.
I am using the vsion 5.4.0.I have a custom resource defined as 
```
public class MyResource extends CustomResource<MyResourceSpec, MyResourceStatus> implements Namespaced, UnknownPropertyPreserving { ... }
```
In my test, I am trying to parse yaml into the object of this class. So I am doing similar
```
class TestClass<T extends HasMetadata,
        L extends KubernetesResource & KubernetesResourceList,
        R extends Resource<T>> {

    protected T copyResource(T resource) {
            ObjectMapper objectMapper = new ObjectMapper();
            try {
                ByteArrayOutputStream baos = new ByteArrayOutputStream();
                objectMapper.writeValue(baos, resource);
                return (T) objectMapper.readValue(baos.toByteArray(), resource.getClass());
            } catch (IOException e) {
                return null;
            }
    }
}
```
However, the mapper writes an object with a duplicate `kind` property. For example: 
```
---
apiVersion: "my.org/v1alpha1"
kind: "MyResource"
metadata:
  name: "my-resouce"
  namespace: "default"
spec:
  property: value
kind: "MyResource"
```
I think kubernetes API does cope with duplicities in the resource yaml, however I do have issues in my test suite. Do you have any idea what has been changed in this regard? The version of jackson has not changed, so it has to be something about `CustomResource` class. Thank you

**Comment by sknot-rh** [comment]
I have created a minimal reproducer. 
[reproducer.zip](https://github.com/fabric8io/kubernetes-client/files/6582590/reproducer.zip)

**Comment by sknot-rh** [comment]
With previous version (5.3.1) it worked :( I think the project is overriding the methods because of some historical context and maybe some additional annotations to provide data for auto-generated documentation. I have removed the overrides and it gets even worse as the apiversion is duplicated too :(

Updated reproducer
[reproducer.zip](https://github.com/fabric8io/kubernetes-client/files/6583705/reproducer.zip)

**Comment by metacosm** [comment]
For some reason, `kind` and `apiVersion` get added to the additional properties… Not sure why at this point.

**Comment by sknot-rh** [comment]
addind this method did remove kind from additional properties
```
    @Override
    public void setKind(String kind) {
        this.kind = kind;
    }
    ```

**Comment by sknot-rh** [comment]
Declaring the `kind` property in the resource did fix the problem. I know it is rather a workaround. I should clean up my code :open_mouth:

**Comment by sknot-rh** [comment]
Hi, I did further debugging of this issue. 
The `kind` and `apiVersion` properties have been annotated by `@JsonProperty(access = Access.READ_ONLY)` in the latest release. 
That has an effect in jackson dependency in `com.fasterxml.jackson.databind.deser.BeanDeserializerFactory#addBeanProps`
this condition is evaluated to be false `if (propDef.hasSetter()) {...}`
so this properties are not added to `_beanProperties`.
`_beanProperties` is used in `com.fasterxml.jackson.databind.deser.BeanDeserializer#vanillaDeserialize` as a list of "known" properties.
Unknown properties are handled by `handleUnknownVanilla(p, ctxt, bean, propName);` 
That results in adding these two properties to the `unknownProperties`.

The question is whether `CustomResource` is right to use `READ_ONLY`, given that we can't know whether the subclass has any `@JsonAnySetter`.

**Comment by metacosm** [comment]
I'm re-opening this issue because we need to address this in some way, either by providing a clean work-around or by documenting the issue.

**Comment by manusa** [comment]
Should be fixed by #3574.

I'm closing the issue, please reopen or create a new one in case the issue persists.
